dist/linux/build.sh

```diff
@@ -0,0 +1,125 @@
+#! /bin/bash
+
+. dist/linux/travis-utils.sh
+. dist/ansi_color.sh
+disable_color
+
+echo "$0" "$@"
+
+# Stop in case of error
+set -e
+
+# Transform long options to short ones
+for arg in "$@"; do
+  shift
+  case "$arg" in
+      "--color"|"-color")   set -- "$@" "-c";;
+      "--build"|"-build")   set -- "$@" "-b";;
+      "--pkg"|"-pkg")       set -- "$@" "-p";;
+      "--gpl"|"-gpl")       set -- "$@" "-g";;
+    *) set -- "$@" "$arg"
+  esac
+done
+# Parse args
+while getopts ":b:p:cg" opt; do
+  case $opt in
+    c) enable_color;;
+    b) BLD=$OPTARG ;;
+    p) PKG_NAME=$OPTARG;;
+    g) ISGPL=true;;
+    \?) printf "$ANSI_RED[GHDL - build] Invalid option: -$OPTARG $ANSI_NOCOLOR\n" >&2
+	exit 1 ;;
+    :)  printf "$ANSI_RED[GHDL - build] Option -$OPTARG requires an argument. $ANSI_NOCOLOR\n" >&2
+	exit 1 ;;
+  esac
+done
+
+rm -f build_ok
+
+#--- Env
+
+echo "travis_fold:start:env.docker"
+printf "$ANSI_YELLOW[Info] Environment $ANSI_NOCOLOR\n"
+env
+echo "travis_fold:end:env.docker"
+
+#--- GPL: gpl-ize sources
+
+if [ "$ISGPL" = "true" ]; then
+    echo "travis_fold:start:gpl.src"
+    printf "$ANSI_YELLOW[Source] create GPL sources $ANSI_NOCOLOR\n"
+    files=`echo *`
+    make -f Makefile.in srcdir=. clean-pure-gpl
+    mkdir ${PKG_NAME}
+    cp -pdrl $files ${PKG_NAME}
+    tar -zcf "${PKG_NAME}.tar.gz" ${PKG_NAME}
+    PKG_NAME="${PKG_NAME}-${BLD}"
+    echo "travis_fold:end:gpl.src"
+fi
+
+#--- Configure
+
+echo "travis_fold:start:configure"
+printf "$ANSI_YELLOW[GHDL - build] Configure $ANSI_NOCOLOR\n"
+
+CDIR=$(pwd)
+export prefix="$CDIR/install-$BLD"
+mkdir "$prefix"
+mkdir "build-$BLD"
+cd "build-$BLD"
+
+case "$BLD" in
+    mcode)
+	      config_opts=""
+        DEP_CLANG=""
+    ;;
+    llvm)
+	      config_opts="--with-llvm-config"
+        DEP_CLANG="clang"
+    ;;
+    llvm-*)
+	      arg_llvm=`find /usr/bin -name llvm-config* | grep -oP "llvm-config.*"`
+	      arg_cxx=clang++`echo $arg_llvm | grep -oP "llvm-config\K.*"`
+	      config_opts="--with-llvm-config=$arg_llvm CXX=$arg_cxx"
+        DEP_CLANG="`$(basename $(find /usr/bin -name clang++*)) --version | grep 'clang'`"
+    ;;
```

Comment:
Honestly I am not sure that llvm-* is a good idea: it makes the code unreadable.  We cannot anymore tell to a user: 'look at the script to see the options we used to configure'.

Reply:
The main point is that we accept any version of LLVM after 3.5 (i.e., 3.5, 3.8, 3.0, 4.0 and 5.0), not just some of them. This is stated in the docs. Having 6 different cases might make the user think that there is some difference between them. Indeed, it requires users with newer versions to edit the file. Conversely, having all the variants with only name differences handled in the same case makes it explicit that we make exactly the same steps, independently of the names of the libraries.

I am ok with reverting these changes, and adding a new case for 5.0 (which is what I did at first for ubuntu18). However, I'd like you to consider if it is worth just adding a comment here:

```
llvm-*)
    # If the version is appended, we just filter it to produce, e.g. for version 3.5:
    # config_opts="--with-llvm-config=llvm-config-3.5 CXX=clang++" ;;

    arg_llvm=`find /usr/bin -name llvm-config* | grep -oP "llvm-config.*"`
```

Reply:
So, let's create a special case for llvm-3.5, and for later version extract the version from the name.  Find won't work if you have multiple llvm installation.

Reply:
Then, lets just split this to a separate PR and have this solved in the 'verbose' way here.

Reply:
Ok.